diff --git dwm/dwm.c dwm/dwm.c
index 5ce3862..55331f6 100644
--- dwm/dwm.c
+++ dwm/dwm.c
@@ -974,12 +974,14 @@ drawstatusbar(Monitor *m, int bh, char* stext, int xx) {
 
 void
 drawbar(Monitor *m) {
-	int x, xx, w;
-	unsigned int i, occ = 0, urg = 0;
+	int x, xx, w, tw, mw;
+	unsigned int i, occ = 0, urg = 0, n = 0, extra = 0;
 	Client *c;
 
 	resizebarwin(m);
 	for(c = m->cl->clients; c; c = c->next) {
+		if(ISVISIBLE(c, c->mon))
+			n++;
 		occ |= c->tags;
 		if(c->isurgent)
 			urg |= c->tags;
@@ -1006,15 +1008,40 @@ drawbar(Monitor *m) {
 	}
 	if((w = x - xx) > bh) {
 		x = xx;
-		if(m->sel) {
-			drw_setscheme(drw, m == selmon ? &scheme[SchemeSel] : &scheme[SchemeNorm]);
-			drw_text(drw, x, 0, w, bh, m->sel->name, 0);
-			drw_rect(drw, x, 0, w, bh, m->sel->isfixed, m->sel->isfloating, 0);
-		}
-		else {
-			drw_setscheme(drw, &scheme[SchemeNorm]);
-			drw_text(drw, x, 0, w, bh, NULL, 0);
+		if(n > 0) {
+			tw = m->sel->name ? TEXTW(m->sel->name) : 0;
+			mw = (tw >= w || n == 1) ? 0 : (w - tw) / (n - 1);
+
+			i = 0;
+			for(c = m->cl->clients; c; c = c->next) {
+				if(!ISVISIBLE(c, c->mon) || c == m->sel)
+					continue;
+				tw = TEXTW(c->name);
+				if(tw < mw)
+					extra += (mw - tw);
+				else
+					i++;
+			}
+			if(i > 0)
+				mw += extra / i;
+
+			for(c = m->cl->clients; c; c = c->next) {
+				if(!ISVISIBLE(c, c->mon))
+					continue;
+				xx = x + w;
+				tw = TEXTW(c->name);
+				w = MIN(m->sel == c ? w : mw, tw);
+
+				drw_setscheme(drw, m->sel == c ? &scheme[SchemeSel] : &scheme[SchemeNorm]);
+				drw_text(drw, x, 0, w, bh, c->name, 0);
+				drw_rect(drw, x, 0, w, bh, c->isfixed, c->isfloating, 0);
+
+				x += w;
+				w = xx - x;
+			}
 		}
+		drw_setscheme(drw, &scheme[SchemeNorm]);
+		drw_text(drw, x, 0, w, bh, NULL, 0);
 	}
 	drw_map(drw, m->barwin, 0, 0, m->ww, bh);
 }
