diff -u -uN dwm/focusurgent.c dwm/focusurgent.c
--- dwm/focusurgent.c	1970-01-01 01:00:00.000000000 +0100
+++ dwm/focusurgent.c	2015-03-19 23:19:25.794530544 +0000
@@ -0,0 +1,26 @@
+void
+focusurgent(const Arg *arg) {
+	Client *c;
+	Monitor *m;
+	unsigned int i;
+
+	for(m = mons; m; m = m->next) {
+		for(c = m->cl->clients; c; c = c->next) {
+			if(c->isurgent) {
+				selmon = m;
+				if(!ISVISIBLE(c, c->mon)) {
+					selmon->seltags ^= 1;
+					selmon->tagset[selmon->seltags] = c->tags;
+					selmon->prevtag = selmon->curtag;
+					for(i=0; !(c->tags & 1 << i); i++);
+					selmon->curtag = i + 1;
+					selmon->lt[selmon->sellt] = selmon->lts[selmon->curtag];
+				}
+				focus(c);
+				arrange(selmon);
+				return;
+			}
+		}
+	}
+}
+
diff --git dwm/config.def.h dwm/config.def.h
index c420699..a939e87 100644
--- dwm/config.def.h
+++ dwm/config.def.h
@@ -85,6 +85,7 @@ static const char *termcmd[]  = { "x-terminal-emulator", NULL };
 #include "focusmaster.c"
 #include "shifttags.c"
 #include "keepfloatingposition.c"
+#include "focusurgent.c"
 static Key keys[] = {
 	/* modifier                     key        function        argument */
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
@@ -100,6 +101,7 @@ static Key keys[] = {
 	{ MODKEY,                       XK_l,      setmfact,       {.f = +0.05} },
 	{ MODKEY,                       XK_Return, zoom,           {0} },
 	{ Mod1Mask,                     XK_Tab,    swapfocus,      {0} },
+	{ MODKEY,                       XK_u,      focusurgent,    {0} },
 	{ MODKEY,                       XK_Tab,    view,           {0} },
 	{ MODKEY|ShiftMask,             XK_c,      killclient,     {0} },
 	{ MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
